### PR TITLE
[INVE-11658] Popup improvements

### PR DESCRIPTION
**Pull request by jaroslawjanas**
### Changes
- Extracted `bindTooltip`, `showTooltip` and `createClusterGeohashPolygon` to a separate file `bind_popup.js`
- Popups now won't re-open when hovering from a popup onto the same polygon it was created by

### Testing
- Tested on Stored Map Layers (Points, Lines and Polygons)
storedLayersTesting
- Tested on WFS Layers (Points and Polygons)
WFSLayersTesting
- Played around with them in general

### Additional Notes
Point popups had to be fixed, were not closing properly when hovering from a point onto a polygon.

### CODE REVIEW

File: public/vislib/vector_layer_types/EsLayer.js [hunk 1]
```diff
@@ -312,9 +226,11 @@ export default class EsLayer {
     const feature = L.marker(
       toLatLng(hitCoords),
       {
+        className: 'point-popup',
         icon: searchIcon(options.icon, options.color, options.size, overlap),
         pane: 'overlayPane'
       });
+    _.set(feature, 'feature.id', hit._id);
```
Comment (corrigancd) [inline public/vislib/vector_layer_types/EsLayer.js:233]: Does feature.id already exist? As in, could it be possible that someone would like to display that field on a popup? 

Set is nice for very nested stuff, in general the consensus seems to be that if it's simple to do in plain javascript, just use it

```suggestion
    feature.id = hit._id;
```

File: public/vislib/vector_layer_types/bind_popup.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  let isPoint;
+  const KEEP_POPUP_OPEN_CLASS_NAMES = ['leaflet-popup', 'tooltip'];
+  let clusterPolygon;
+  const keepPopupOpen = (target) => {
+    const popupType = utils.getParent(target, KEEP_POPUP_OPEN_CLASS_NAMES);
+    let popupClassnameCheck;
+    if (popupType && popupType.className) {
+      popupClassnameCheck = popupType.className.includes(mouseoverId);
+    }
+    else if(isPoint) {
```
Comment (corrigancd) [inline public/vislib/vector_layer_types/bind_popup.js:60]: If you hover outside of a point popup, I think that means the mouse will either not be on the point (popup should disappear) or you hover back onto the point (popup should remain open). So I'm wondering if the same logic can be applied to point, lines and polygons?

File: public/vislib/vector_layer_types/vector.js [hunk 3]
```diff
@@ -129,71 +131,6 @@ export default class Vector {
     return layer;
   };
 
-
-
-  /**
- * Binds popup and events to each feature on map
- *
- * @method bindPopup
```
Comment (corrigancd) [inline public/vislib/vector_layer_types/vector.js:137]: :sunglasses:

File: public/vislib/vector_layer_types/vector.js [hunk 4]
```diff
@@ -202,8 +139,11 @@ export default class Vector {
     const feature = L.marker(
       toLatLng(hit.geometry.coordinates),
       {
+        className: 'point-popup',
         icon: markerIcon(options.color, options.size)
       });
+    _.set(feature, 'feature.id', hit.id);
```
Comment (corrigancd) [inline public/vislib/vector_layer_types/vector.js:145]: same as above suggestion

File: public/vislib/vector_layer_types/EsLayer.js (hunk 1 above)
Comment (jaroslawjanas) [reply public/vislib/vector_layer_types/EsLayer.js:233]: Then it should be ` feature.feature = { id: hit._id };` I think

File: public/vislib/vector_layer_types/vector.js (hunk 4 above)
Comment (jaroslawjanas) [reply public/vislib/vector_layer_types/vector.js:145]: Then it should be ` feature.feature = { id: hit._id };` I think

File: public/vislib/vector_layer_types/bind_popup.js (hunk 2 above)
Comment (jaroslawjanas) [reply public/vislib/vector_layer_types/bind_popup.js:60]: Some logic to handle points is needed.
When hovering from point to polygon `true` is returned for `keepPopupOpen` either different logic is needed for that function or points and polygons need to be separated.

**Comment by jaroslawjanas** [comment]
POIs
whyDontYouCheckMyPointOfInterest^^

**Comment by jaroslawjanas** [comment]
AintTheWorldMadeUpOfPolygons

It **"softly"** brakes after  
**Pol-1** -> **Pop(1)** -> **Pol-2** -> **Pop(2)** -> **Pol-1** -> **No Pop** -> **Pol-2** -> **No pop** 
then it recovers.

**Comment by jaroslawjanas** [comment]
Also when the out event goes straight to browser or outside browser element this happens
thisIsAllForResearchPurposes

**FIXED**

**Comment by corrigancd** [comment]
This is better now?  
PopupsMightBeFixedThisTime

**Comment by jaroslawjanas** [comment]
Ah I see what you did there.
Tested, looks good. Merging, and hopefully it won't have to be touched again for a very very long time.

**Pull request merged**